ethereum/statetransition/src/testFixtures/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodyStand.java

```diff
@@ -0,0 +1,209 @@
+
+  public final SpecConfigEip7594 config;
+  public final DataColumnSidecarCustodyImpl custody;
+
+  public final DataStructureUtil dataStructureUtil;
+
+  private final List<SlotEventsChannel> slotListeners = new CopyOnWriteArrayList<>();
+  private final List<FinalizedCheckpointChannel> finalizedListeners = new CopyOnWriteArrayList<>();
+  private final int totalCustodySubnetCount;
+
+  private UInt64 currentSlot = UInt64.ZERO;
+
+  public DasCustodyStand(
+      Spec spec, UInt64 currentSlot, UInt256 myNodeId, int totalCustodySubnetCount) {
+    this.spec = spec;
+    this.myNodeId = myNodeId;
+    this.blockResolver = new CanonicalBlockResolverStub(spec);
+    this.config = SpecConfigEip7594.required(spec.forMilestone(SpecMilestone.EIP7594).getConfig());
+    this.custody =
+        new DataColumnSidecarCustodyImpl(
+            spec, blockResolver, db, myNodeId, totalCustodySubnetCount);
+    subscribeToSlotEvents(this.custody);
+    subscribeToFinalizedEvents(this.custody);
+
+    DataStructureUtil util = new DataStructureUtil(0, spec);
+    BLSSignature singleSignature = util.randomSignature();
+    BLSPublicKey singlePubKey = util.randomPublicKey();
+    this.dataStructureUtil =
+        util.withSignatureGenerator(__ -> singleSignature).withPubKeyGenerator(() -> singlePubKey);
+    this.totalCustodySubnetCount = totalCustodySubnetCount;
+  }
+
+  public SignedBeaconBlock createBlockWithBlobs(int slot) {
+    return createBlock(slot, 3);
+  }
+
+  public SignedBeaconBlock createBlockWithoutBlobs(int slot) {
+    return createBlock(slot, 0);
+  }
+
+  public SignedBeaconBlock createBlock(int slot, int blobCount) {
+    UInt64 slotU = UInt64.valueOf(slot);
+    BeaconBlockBody beaconBlockBody =
+        dataStructureUtil.randomBeaconBlockBodyWithCommitments(blobCount);
+    BeaconBlock block = dataStructureUtil.randomBeaconBlock(slotU, beaconBlockBody);
+    return dataStructureUtil.signedBlock(block);
+  }
+
+  public DataColumnSidecar createSidecar(SignedBeaconBlock block, int column) {
+    return dataStructureUtil.randomDataColumnSidecar(block.asHeader(), UInt64.valueOf(column));
+  }
+
+  public boolean hasBlobs(BeaconBlock block) {
+    return block
+        .getBody()
+        .toVersionEip7594()
+        .map(b -> !b.getBlobKzgCommitments().isEmpty())
+        .orElse(false);
+  }
+
+  public Collection<UInt64> getCustodyColumnIndexes(UInt64 slot) {
+    UInt64 epoch = spec.computeEpochAtSlot(slot);
+    return spec.atEpoch(epoch)
+        .miscHelpers()
+        .toVersionEip7594()
+        .map(
+            miscHelpersEip7594 ->
+                miscHelpersEip7594.computeCustodyColumnIndexes(myNodeId, totalCustodySubnetCount))
+        .orElse(Collections.emptyList());
+  }
+
+  public int getCustodyPeriodSlots() {
+    int custodyPeriodEpochs = config.getCustodyRequirement();
+    int epochSlots = spec.slotsPerEpoch(UInt64.ZERO);
+    return custodyPeriodEpochs * epochSlots;
+  }
+
+  public UInt64 getMinCustodySlot() {
```

Comment:
it should be whole epochs as per spec

Reply:
I want slots :))

Reply:
Yep, you are right, we need to round to the epoch boundary